govcd/vm.go

```diff
@@ -1508,7 +1530,58 @@ func (vm *VM) UpdateComputePolicy(computePolicy *types.VdcComputePolicy) (*VM, e
 
 }
 
+// UpdateComputePolicyV2Async updates VM Compute policy with the given compute policies using v2.0.0 OpenAPI endpoint,
+// and returns a Task and an error. Updating with an empty compute policy ID will remove it from the VM. Both
+// policies can't be empty as the VM requires at least one policy.
+func (vm *VM) UpdateComputePolicyV2Async(sizingPolicyId, placementPolicyId string) (Task, error) {
```

Comment:
I've got one concern about function parameters - what will we do when we get to vGpu policy. I think leaving the function as it is will force us to deprecate it. Should we better accept the container of all policies `*types.ComputePolicy`?

Reply:
That is a good point! The only thing I dislike is that it will be a bit more verbose in the consumer side (Terraform provider in this case) but it will be 100% complete indeed, which is nice as it will avoid deprecations as you say.

I will change it :+1:

PS: Another option would be having a variadic function, like

```
UpdateComputePolicyV2Async(policyIds... string)
```

but we lose which policy type each one is, unless we comment it.

Reply:
I added a `vGpuPolicyId` parameter, although it is ignored. I couldn't find it is used in the [documentation](https://developer.vmware.com/apis/1260/vmware-cloud-director/doc/doc//types/VmType.html).